cmd/oras/root/manifest/index/create.go

```diff
@@ -188,7 +193,8 @@ func getPlatform(ctx context.Context, target oras.ReadOnlyTarget, manifestBytes
 	return &platform, nil
 }
 
-func pushIndex(ctx context.Context, target oras.Target, desc ocispec.Descriptor, content []byte, ref string, extraRefs []string, path string, printer *output.Printer) error {
+func pushIndex(ctx context.Context, onIndexPushed func(path string) error, onTagged func(desc ocispec.Descriptor, tag string) error,
+	target oras.Target, desc ocispec.Descriptor, content []byte, ref string, extraRefs []string, path string) error {
```

Comment:
Try using handlers like:
```suggestion
func pushIndex(ctx context.Context, pmh metadata.ManifestPackHandler, th metadata.TaggedHandler,target oras.Target, desc ocispec.Descriptor, content []byte, ref string, extraRefs []string, path string) error {
```

Reply:
This function signature is now `func pushIndex(ctx context.Context, displayStatus status.ManifestIndexCreateHandler, taggedHandler metadata.TaggedHandler, target oras.Target, desc ocispec.Descriptor, content []byte, ref string, extraRefs []string, path string)` after resolving this [comment](https://github.com/oras-project/oras/pull/1509#discussion_r1796514181). Still long param list.